apps/antalmanac/src/components/Map/Routes.tsx

```diff
@@ -0,0 +1,141 @@
+     */
+    fitSelectedRoutes: false,
+
+    plan: L.Routing.plan(waypoints, {
+      addWaypoints: false,
+      createMarker: () => false,
+    }),
+
+    routeLine(route) {
+      const line = L.Routing.line(route, {
+        addWaypoints: false,
+        extendToWaypoints: true,
+        missingRouteTolerance: 0,
+        styles: [
+          { color: 'skyblue', opacity: 0.5, weight: 30 }, // invisble line extends the range of click/hover events
+          { color: props.color, weight: 3 },
+        ],
+      })
+
+      const totalTime = route.summary?.totalTime || 0
+      const totalDistance = route.summary?.totalDistance || 0
+
+      const duration = totalTime > 30 ? `${Math.round(totalTime / 60)} min` : '<1 min'
+      const miles = `${Math.floor(totalDistance / 1.609 / 10) / 100} mi`
+
+      const content = `
+       <div style="position:relative; 
+                  top: -200%;
+                  left:2px;
+                  pointer-events: none;
+                  background-color: white;
+                  border-left-color: ${props.color};
+                  border-left-style: solid;
+                  width: fit-content;
+                  border-left-width: 5px;
+                  padding-left: 10px;
+                  padding-right: 10px;
+                  padding-top: 4px;
+                  padding-bottom: 4px;"
+       />
+       <span style="color:${props.color}">${duration}</span>
+       <br>
+       <span style="color:#888888">${miles}</span>
+      `
+      const popup = L.popup({ content })
+
+      /**
+       * @see {@link https://github.com/perliedman/leaflet-routing-machine/issues/117}
+       */
+      line.eachLayer((lineLayer) => {
+        lineLayer.on('click', (leafletMouseEvent) => {
+          popup.setLatLng(leafletMouseEvent.latlng).openOn(context.map)
+        })
+        lineLayer.on('mouseover', (leafletMouseEvent) => {
+          popup.setLatLng(leafletMouseEvent.latlng).openOn(context.map)
+        })
+        lineLayer.on('mousemove', (leafletMouseEvent) => {
+          popup.setLatLng(leafletMouseEvent.latlng).openOn(context.map)
+        })
+      })
+
+      return line
+    },
+  })
+
+  const router = createElementObject(routerControl, context)
+  return router
+}
+
+/**
+ * Turn the createRouter function into a hook for the main component.
+ */
+const useRouter = createElementHook(createRouter)
+
+/**
+ * Given waypoints of a route and a color for the route, draw a route to the map.
+ * Forward the props to a custom hook that manages the leaflet element's lifecycle.
+ */
+export default function CourseRoutes(props: Props) {
+  const map = useMap()
+  const context = useLeafletContext()
+  const routerRef = useRouter(props, context)
+
+  useEffect(() => {
+    routerRef.current.instance.addTo(map)
+    routerRef.current.instance.hide()
+    return () => {
+      routerRef.current.instance.remove()
+    }
+  }, [])
+
+  return null
```

Comment:
This is supposedly the correct strategy to use react-leaflet, but it doesn't seem to do anything as far as I can tell. I thought it would help with the route ghosting that occurs, but it doesn't.

Reply:
I would just include a comment that links to the source if someone's confused about why it's done this way

Reply:
I mention it on line 27; should I repeat or move this info?

Reply:
Oh yeah. That's good then